### PR TITLE
PDM_PCM-read, irq and set gain function implementation

### PR DESCRIPTION
**Pull request by NikhitaR-IFX**
### Summary

In this PR, all the below functions are implemented but yet to be fully tested.
- PDM_PCM.irq()
- PDM_PCM.readinto()
- PDM_PCM.set_gain()

In the source file, you will find 4 sections:
1. PDM_PCM higher level MPY functions --> at some point this should move to extmod/ and the functions in this are port independent
2. Port-specific private functions for DMA and PDM_PCM support 
3. MPY bindings for the module
4. Implementation to support stream protocol

### Test Status:
All the testin so far is being done manually and is later when successful to be added to HIL.

- PDM_PCM.set_gain() atleast does the job of setting the values and this is ensured as cyhal api is supposed to throw an error if we try to set unacceptable values.
- PDM_PCM.irq() sets the callback fine.
- PDM_PCM.readinto() is waiting for the interrupt trigger forever. I suspect the buffer configurations.

**Note**: Based on further testing, I feel changes would happen only to following functions and you may ignore if needed while reviewing the same:
- fill_appbuf_from_ringbuf()
- fill_appbuf_from_ringbuf_non_blocking()

Also I am aware there are printf()/mp_printf() statements in the beginning of every function, but only to help with debugging. This will be replaced later by mplogger_print().

### CODE REVIEW

File: docs/psoc6/quickref.rst [hunk 1]
```diff
@@ -633,19 +633,31 @@ Constructor
 Methods
 -------
```
Comment (jaenrig-ifx) [inline docs/psoc6/quickref.rst:634]: The headlines need to have 1 level lower (Methods, Constructor, etc.)

Comment (jaenrig-ifx) [reply docs/psoc6/quickref.rst:634]: And to keep consistency with the rest of the classes, I would name this "PDM-PDM bus"

**Review by jaenrig-ifx — APPROVED** [review]
All good, with the structure and the API :) 

Only the comments on the documentation. Which is generated here:
https://ifx-micropython.readthedocs.io/en/pdm-read-blocking/psoc6/quickref.html#pdm-pcm

Without tests passing, I will postpone a deeper of the code functionality.

**Review by IFX-Anusha — APPROVED** [review]
Looks Good!

File: docs/psoc6/quickref.rst (hunk 1 above)
Comment (NikhitaR-IFX) [reply docs/psoc6/quickref.rst:634]: Noted

**Review by ramya-subramanyam — APPROVED** [review]
Looks good!

**Pull request closed without merging**